GraphsV2/GaugeChart/index.js

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { compose } from 'redux';
+import { Cell, PieChart, Pie } from 'recharts';
+
+import WithConfigHOC from '../../HOC/WithConfigHOC';
+import WithValidationHOC from '../../HOC/WithValidationHOC';
+import config from './default.config';
+import colorConvert from 'color-convert';
+
+const GaugeChart = (props) => {
+
+    const {
+        data: originalData,
+        width,
+        height,
+        properties,
+    } = props;
+
+    const {
+        minValue,
+        maxValue,
+        gaugeTicks,
+        fontSize,
+    } = properties;
+    
+    const RADIAN = Math.PI / 180;
+    
+    const chartValue = originalData[0].value;
+    
+    const chartData = [];
+    
+    let gaugeTickValue = parseInt(parseInt(maxValue) / parseInt(gaugeTicks));
+    
+    for (let i = parseInt(minValue) + gaugeTickValue; i <= parseInt(maxValue); i += gaugeTickValue) {
+        chartData.push({
+            name: i,
+            value: gaugeTickValue,
+            label: i,
+            color: ((code) => {
+              var rgb = colorConvert.hsl.rgb(code, 100, 50);
+              return 'rgb(' + rgb[0] + ',' + rgb[1] + ',' + rgb[2] + ')';
+            })(parseInt(maxValue) - i),
+        })
+    }
+
+    const sumValues = chartData
```

Comment:
is this required for every re-render I would say only when charData is changed so why not save the computation?